rig/machine_control/machine_controller.py

```diff
@@ -307,6 +332,63 @@ def boot(self, width, height,
         return True
 
     @ContextMixin.use_contextual_arguments()
+    def discover_connections(self, x=0, y=0):
```

Comment:
I'd been wondering when/how/if to call it automatically and couldn't think of a good place. I'm not fond of something like get_system_info having such a "magic" side-effect. Any other ideas?